dagstore_control.go

```diff
@@ -124,11 +126,13 @@ func (d *DAGStore) control() {
 				// optimistically increment the refcount to acquire the shard. The go-routine will send an `OpShardRelease` message
 				// to the event loop if it fails to acquire the shard.
 				s.refs++
+				// TODO Can we pass in  a looping pointer variable as is to a go-routine ?
```

Comment:
@raulk

Reply:
This is fine. The pointer is passed by value. A different thing would be if we closed over the value, in which case it would be racy, e.g.:

```go
for _, s := range ... {
    go func() {
       // do something with s
    }()
}
```

Take a look at this: https://eli.thegreenplace.net/2019/go-internals-capturing-loop-variables-in-closures/.

Reply:
When in doubt, you can cook up a little Go Playground thing: https://play.golang.org/p/TogtEsZAkiY